compiler/src/iree/compiler/Codegen/Common/EncodingUtils.cpp

```diff
@@ -100,6 +102,40 @@ static RankedTensorType transposeIfNarrowNResult(RankedTensorType tensorType) {
   return RankedTensorType::get(newShape, elemType, newEncoding);
 }
 
+template <typename T>
+SmallVector<T> getTileSwizzlingDims(ArrayRef<T> typeShape,
```

Comment:
Are you going to use this refactored function for below two `resolve*` implementations?

Reply:
yup, combined everything together